dipy/reconst/sfm.py

```diff
@@ -420,7 +420,24 @@ def design_matrix(self):
         return sfm_design_matrix(self.gtab, self.sphere, self.response,
                                  'signal')
 
-    def fit(self, data, mask=None):
+    def fit_solver2voxels(self, isopredict, vox_data, vox, parallel=False):
+        # In voxels in which S0 is 0, we just want to keep the
+        # parameters at all-zeros, and avoid nasty sklearn errors:
+        if not (np.any(~np.isfinite(vox_data)) or np.all(
+                vox_data == 0)):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+
+        if parallel is True:
```

Comment:
I recommend avoiding using `is`.  It is confusing for many people. for example
`if 1 is True` returns `False` when many people expect `True` due to a misunderstanding of `is` .

You can do `if parallel:` instead

Reply:
Sounds good to me, assuming that `parallel` will always be set to either `True` or `False` and not `None` or some other number/character(s) (e.g. `5`) ...  I've been in the (bad) habit of using `is` recently to avoid the latter scenario, which can cause issues in nipype. I bet we can at least both agree against `parallel == True` haha

Reply:
`parallel == True` not really pythonic 😄 ... if the type is important, you can use `isinstance`

`if parallel and isinstance(parallel, bool):`